### PR TITLE
Spotify

### PR DESCRIPTION
**Pull request by ArgoNavis1763**
Spotify endpoints set up

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,32 +6,34 @@ import fs from "fs";
...
+app.use(express.json());
 
 app.get("/", (req, res) => {
   res.send("Hello World!");
 });
 
 const storage = multer.diskStorage({
   filename: function (req, file, cb) {
-    console.log("filename");
+    // console.log("filename");
```
Comment (TomGroombridge) [inline index.js]: remove log

File: index.js [hunk 2]
```diff
@@ -6,32 +6,34 @@ import fs from "fs";
...
     cb(null, file.originalname);
   },
   destination: function (req, file, cb) {
-    console.log("storage");
+    // console.log("storage");
```
Comment (TomGroombridge) [inline index.js]: remove log

File: index.js [hunk 3]
```diff
@@ -6,32 +6,34 @@ import fs from "fs";
...
     cb(null, "./uploads");
   },
 });
 
 const upload = multer({ storage });
 app.post("/audio_info", upload.single("file"), async (req, res) => {
-  console.log("req.file", req.file);
+  // console.log("req.file", req.file);
```
Comment (TomGroombridge) [inline index.js]: remove log

File: index.js [hunk 4]
```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
   }
 });
 
+app.post("/spotify-auth", async (req, res) => {
+  // console.log("request.body", req.body);
```
Comment (TomGroombridge) [inline index.js]: remove

File: index.js [hunk 5]
```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
...
+      data: qs.stringify(payload),
+      headers: {
+        Authorization: `Basic ${process.env.CLIENT_DETAILS}`,
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    });
+    res.send(result.data);
+    console.log("result.data", result.data);
+  } catch (error) {
+    console.log(error);
```
Comment (TomGroombridge) [inline index.js]: if there is an error I think we should be sending a 500 response, at the moment we are just logging

File: index.js [hunk 6]
```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
...
+
+app.get(`/artist/:id/top-tracks`, async (req, res) => {
+  const id = req.params.id;
+ try {
+  
+  const result = await axios({
+    method: "get",
+    url: `https://api.spotify.com/v1/artists/${id}/top-tracks?market=GB`,
+    headers: {
+      Authorization: `Bearer ${process.env.SPOTIFY_BEARER_TOKEN}`,
```
Comment (TomGroombridge) [inline index.js]: should this be a env or should this be passed as a header in the request?

File: index.js [hunk 7]
```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
...
+
+app.get(`/artist/:id/related-artists`, async (req, res) => {
+  const id = req.params.id;
+ try {
+  
+  const result = await axios({
+    method: "get",
+    url: `https://api.spotify.com/v1/artists/${id}/related-artists`,
+    headers: {
+      Authorization: `Bearer ${process.env.SPOTIFY_BEARER_TOKEN}`,
```
Comment (TomGroombridge) [inline index.js]: again should this be passed as a header not a ENV?

File: index.js [hunk 8]
```diff
@@ -51,6 +53,71 @@ app.post("/audio_info", upload.single("file"), async (req, res) => {
...
+      'Content-Type': 'application/json'
+    }
+
+  })
+  res.send(result.data);
+ } catch (error) {
+  console.log(error)
```
Comment (TomGroombridge) [inline index.js]: again can we make this a 500 response

**Review by TomGroombridge — COMMENTED** [review]
Looks really good just a few comments

File: index.js (hunk 1 above)
Comment (ArgoNavis1763) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (ArgoNavis1763) [reply index.js]: Done

File: index.js (hunk 3 above)
Comment (ArgoNavis1763) [reply index.js]: Done

File: index.js (hunk 4 above)
Comment (ArgoNavis1763) [reply index.js]: Done

File: index.js (hunk 8 above)
Comment (ArgoNavis1763) [reply index.js]: Done

File: index.js (hunk 5 above)
Comment (ArgoNavis1763) [reply index.js]: Done

File: index.js (hunk 6 above)
Comment (ArgoNavis1763) [reply index.js]: So does this not need to be an env file?

File: index.js (hunk 7 above)
Comment (ArgoNavis1763) [reply index.js]: Same as above

**Pull request merged**